tests/outlineCompiler_test.py

```diff
@@ -761,6 +761,30 @@ def test_compile_strange_glyph_order(self, quadufo):
         compiler.compile()
         assert compiler.otf.getGlyphOrder() == EXPECTED_ORDER
 
+    def test_compile_reorder_space_glyph(self, FontClass):
+        """
+        Test that ufo2ft always puts .notdef first, and put space second if no
+        explicit glyph order is set.
+        """
+        # This UFO does not have a "public.glyphOrder"
+        ufo = FontClass(getpath("TestSpaceOrder.ufo"))
+
+        EXPECTED_ORDER = [".notdef", "space", "a", "b", "c", "d", "e", "f", "g"]
+        # We can’t assert a specific glyph order here because ufolib2 and
+        # defcon do things differently, so only assert it is not the expected
```

Comment:
maybe you could concisely write here how the two differ?

Reply:
It seems that ufoLib2 is using the order from the layer’s content.plist, but defcon order seems to be random.

Reply:
you're talking about the order of the ufo.keys() here, not the actual glyphOrder list, right?
yeah, in that case, ufoLib2 is using a dict internally (which remembers insertion order) whereas defcon may be using an unordered set of glyphNames. In any case, better not to rely on the order of the keys